eventhubs-streamanalytics-eventhubs/tools/eh-asa-perfmon/EventHandler.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Azure.Messaging.EventHubs.Consumer;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace StreamingAtScale
+{
+    public class EventReceiver
+    {
+        private readonly object _lock = new object();
+
+        public async Task Receive(string connectionString, CancellationToken _cancellationToken)
+        {
+            await using (var consumerClient = new EventHubConsumerClient(EventHubConsumerClient.DefaultConsumerGroupName, connectionString))
+            {
+                Console.WriteLine("The application will now start to listen for incoming message.");
+
+                ReadEventOptions readOptions = new ReadEventOptions
+                {
+                    MaximumWaitTime = TimeSpan.FromMilliseconds(150)
+                };
+
+                var csvOutput = File.CreateText("./resulttest.csv");
+                csvOutput.WriteLine("EventCount,BatchCount,BatchFrom,BatchTo,MinLatency,MaxLatency,AvgLatency");
+                try
+                {
+                    var events = consumerClient.ReadEventsAsync(startReadingAtEarliestEvent: false, readOptions, _cancellationToken);
+
+                    await foreach (PartitionEvent currentEvent in events)
+                    {
+                        var eventData = currentEvent.Data;
+                        if (eventData == null) continue;
+                        int eventCount = 0;
+                        var listTimeSpan = new List<TimeSpan>();
+                        var listDateTime = new List<DateTimeOffset>();
+
+                        var eventBody = Encoding.UTF8.GetString(eventData.Body.ToArray()).Split('\n');
+
+                        foreach (var bodyInfo in eventBody)
+                        {
+                            try
+                            {
+                                var message = JsonSerializer.Deserialize<Dictionary<string, object>>(bodyInfo);
+                                eventCount += 1;
```

Comment:
nit:  Convention here would normally be `eventCount++`

Reply:
resolved